lib/improver/tests/utilities/test_cube_metadata.py

```diff
@@ -883,5 +885,106 @@ def test_error_not_probability(self):
             extract_diagnostic_name('lwe_precipitation_rate')
 
 
+class Test_generate_hash(IrisTest):
+    """Test utility to generate md5 hash codes from a multitude of inputs."""
+
+    def test_string_input(self):
+        """Test the expected hash is returned when input is a string type."""
+
+        hash_input = 'this is a test string'
+        result = generate_hash(hash_input)
+        expected = "8e502f6a5b4a2e0f226649210895cebc"
+        self.assertIsInstance(result, str)
+        self.assertEqual(result, expected)
+
+    def test_numeric_input(self):
+        """Test the expected hash is returned when input is a numeric type."""
+
+        hash_input = 1000
+        result = generate_hash(hash_input)
+        expected = "d017763f19ef64f920c43fc57413d171"
+        self.assertIsInstance(result, str)
+        self.assertEqual(result, expected)
+
+    def test_dictionary_input(self):
+        """Test the expected hash is returned when input is a dictionary."""
+
+        hash_input = {'one': 1, 'two': 2}
+        result = generate_hash(hash_input)
+        expected = "4735f4a74dd17d27b383de504a87e324"
+        self.assertIsInstance(result, str)
+        self.assertEqual(result, expected)
+
+    def test_dictionary_order_variant(self):
```

Comment:
This might be a silly question / request, but can we also have a demonstration that when you put the same inputs in twice you get the same hash out?  (This is coming from a scientist, where I'm aware we have scientists maintaining this code, who might like me be unaware of exactly how hashes work.)

Reply:
Done.

Reply:
Thanks.